app/parse.py

```diff
@@ -14,9 +29,64 @@ class Product:
     rating: int
     num_of_reviews: int
 
+    def parse_single_product(self, product: Tag) -> Product:
+        return Product(
+            title=product.select_one(".title")["title"],
+            description=product.select_one(".description").text.replace(
+                "\xa0", " "
+            ),
+            price=float(product.select_one("h4.price").text.replace("$", "")),
+            rating=len(product.select(".glyphicon-star")),
+            num_of_reviews=int(
+                product.select_one("p.pull-right").text.split()[0]
+            ),
+        )
+
+    def get_page_products(
+        self, page_url: str, driver: WebDriver
+    ) -> list[Product]:
+        driver.get(page_url)
+        cookies = driver.find_elements(By.CLASS_NAME, "acceptCookies")
+
+        if cookies:
+            cookies[0].click()
+
+        button_more = driver.find_elements(
+            By.CLASS_NAME, "ecomerce-items-scroll-more"
+        )
+
+        if button_more:
+            while button_more[0].is_displayed():
+                button_more[0].click()
+                time.sleep(0.1)
+
+        page = driver.page_source
+        page_soup = BeautifulSoup(page, "html.parser")
+
+        products = page_soup.select(".thumbnail")
+
+        return [
+            self.parse_single_product(self, product) for product in products
+        ]
+
+
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def write_products_to_csv(products: [Product], output_csv_path: str) -> None:
+    with open(output_csv_path, "w", newline="", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows([astuple(product) for product in products])
+
 
 def get_all_products() -> None:
-    pass
+    with webdriver.Chrome() as driver:
+        for name, url in URLS.items():
+            write_products_to_csv(
+                products=Product.get_page_products(Product, url, driver),
```

Comment:
```suggestion
                products=Product.get_page_products(url, driver),
```
After changing to class method